crates/nu-protocol/src/signature.rs

```diff
@@ -309,6 +322,34 @@ impl Signature {
         self
     }
 
+    pub fn add_positional_completion(mut self, completion: Completion) -> Signature {
+        let Self {
+            required_positional,
+            optional_positional,
+            rest_positional,
+            ..
+        } = &mut self;
+        match (
+            rest_positional.as_mut(),
+            optional_positional.as_mut_slice(),
+            required_positional.as_mut_slice(),
+        ) {
+            (Some(rest), _, _) => rest.custom_completion = Some(completion),
+            (_, [.., last], _) | (_, _, [.., last]) => last.custom_completion = Some(completion),
+            _ => panic!("oops no parameters"),
+        }
+        self
+    }
+
+    pub fn add_named_completion(mut self, completion: Completion) -> Signature {
```

Comment:
With this imperative semantics we need to document the behavior.

```suggestion
    /// Add a [`Completion`] to the `.named()`/`.switch()`/`.required_named()` that was last added to the `Signature`
    pub fn add_named_completion(mut self, completion: Completion) -> Signature {
```

Reply:
I don't like the imperative behavior to be honest, just went with it to make the MVP without having to refactor all `Signature` constructions.